app/models/tag.rb

```diff
@@ -48,6 +48,65 @@ def restricted?
     restricted
   end
 
+  # int -> Kaminari::PaginatableArray
+  def recent_edits_for_homepage(page = 1)
+    paginate page, recent_edits(page), (taggings.count * 2)
+  end
+
+  # -> [Hash]
+  # Adds the ActiveRecord Tagable with key 'tagable' to
+  # each hash provided by #recent_edits_query.
+  def recent_edits(page = 1)
+    edits = recent_edits_query(page)
+    active_record_lookup = active_record_lookup_for_recent_edits(edits)
+    edits.map do |edit|
+      edit.tap { |h| h.store 'tagable', active_record_lookup.dig(h['tagable_class'], h['tagable_id']) }
+    end
+  end
+
+  # int -> [ Hash ]
+  # Hash keys: tagging_id, tagable_id, tagable_class, tagging_created_at, event_timestamp, event
+  def recent_edits_query(page = 1)
```

Comment:
wow! learned a lot by reading and understanding this query. :)

Reply:
:+1: